Packs/HelloWorld/Integrations/FeedHelloWorld/FeedHelloWorld.py

```diff
@@ -0,0 +1,437 @@
+                                 url_suffix='',
+                                 full_url=self._base_url,
+                                 resp_type='text',
+                                 )
+
+        result = []
+
+        # In this case the feed output is in text format, so extracting the indicators from the response requires
+        # iterating over it's lines solely. Other feeds could be in other kinds of formats (CSV, MISP, etc.), or might
+        # require additional processing as well.
+        try:
+            indicators = res.split('\n')
+
+            for indicator in indicators:
+                if auto_detect_indicator_type(indicator):
+                    result.append({
+                        'value': indicator,
+                        'type': auto_detect_indicator_type(indicator),
+                        'FeedURL': self._base_url
+                    })
+
+        except ValueError as err:
+            demisto.debug(str(err))
+            raise ValueError(f'Could not parse returned data to Json. \n\nError massage: {err}')
+        return result
+
+
+def test_module(client: Client) -> str:
+    """Builds the iterator to check that the feed is accessible.
+    Args:
+        client: Client object.
+    Returns:
+        Outputs.
+    """
+    # INTEGRATION FEED DEVELOPER TIP
+    # Client class should raise the exceptions, but if the test fails
+    # the exception text is printed to the Cortex XSOAR UI.
+    # If you have some specific errors you want to capture (i.e. auth failure)
+    # you should catch the exception here and return a string with a more
+    # readable output (for example return 'Authentication Error, API Key
+    # invalid').
+    # Cortex XSOAR will print everything you return different than 'ok' as
+    # an error
+
+    client.build_iterator()
+    return 'ok'
+
+
+def fetch_indicators(client: Client, tlp_color: Optional[str] = None, feed_tags: List = [], limit: int = -1) \
+        -> List[Dict]:
+    """Retrieves indicators from the feed
+    Args:
+        client (Client): Client object with request
+        tlp_color (str): Traffic Light Protocol color
+        feed_tags (list): tags to assign fetched indicators
+        limit (int): limit the results
+    Returns:
+        Indicators.
+    """
+    iterator = client.build_iterator()
+    indicators = []
+    if limit > 0:
+        iterator = iterator[:limit]
+
+    # extract values from iterator
+    for item in iterator:
+        value = item.get('value')
+        type_ = item.get('type')
+        raw_data = {
+            'value': value,
+            'type': type_,
+        }
+
+        # create indicator object for each value
+        for key, val in item.items():
+            raw_data.update({key: val})
+        indicator_obj = {
+            'value': value,
+            'type': type_,
+            'service': 'HelloWorld',
+            'fields': {},
+            'rawJSON': raw_data
+        }
+
+        if feed_tags:
```

Comment:
I think it would be valuable to explain better in the comments the "fields" and how they are used, and how to manage custom fields if you have them in your feed.

Reply:
Done